modules/@amperka/octoliner.js

```diff
@@ -9,31 +9,38 @@ var Octoliner = function(opts) {
     opts.i2c = I2C1;
   }
   this.expander = require('@amperka/gpio-expander').connect(opts);
-  this.expander.pwmFreq(60000);
+  this.expander.pwmFreq(8000);
   this._sensePin = 0;
   this._ledBrightnessPin = 9;
+  this.expander.analogWrite(this._ledBrightnessPin, 1);
   this._sensorPinMap = [4, 5, 6, 8, 7, 3, 2, 1];
   this._lineState = [];
   this.setSensitivity(0.91);
-  this.setBrightness(1);
+  this.mapPatternToLine = this.defaultMapPatternToLine;
+  this.mapAnalogToPattern = this.defaultMapAnalogToPattern;
 };
 
 Octoliner.prototype.setSensitivity = function(sense) {
   this.expander.analogWrite(this._sensePin, sense);
 };
 
-Octoliner.prototype.setBrightness = function(brightness) {
-  this.expander.analogWrite(this._ledBrightnessPin, brightness);
-};
+// deprecated
+Octoliner.prototype.setBrightness = function() {};
 
 Octoliner.prototype.analogRead = function(sensor) {
-  sensor &= 0x07;
-  return this.expander.analogRead(this._sensorPinMap[sensor]);
+  return this.expander.analogRead(this._sensorPinMap[sensor & 0x07]);
 };
 
-Octoliner.prototype.digitalRead = function(sensor) {
-  sensor &= 0x07;
-  return this.expander.digitalRead(this._sensorPinMap[sensor]);
+Octoliner.prototype.analogReadAll = function(analogArray) {
+  for (var i = 0; i < 8; i++) {
+    analogArray[i] = this.analogRead(i);
+  }
```

Comment:
No reason to do this in JS. JavaScript memory is managed, so just create an array and return it.

Like (not checked):

```js
var values = [0, 0, 0, 0, 0, 0, 0, 0];
for (var i = 0; i < 8; i++) {
  values[i] = this.analogRead(i);
}

return values;
```

Reply:
It is right. I am tryed to save the same interface.

Reply:
Done. 
Do not forget to indicate the difference in the version of the API for JS.